File: elixir/lib/formhandle/commands/open.ex
defmodule FormHandle.Commands.Open do
  alias FormHandle.Output

  @url "https://formhandle.dev/swagger/"

  def run(opts) do
    if opts[:json] do
      Output.json(%{"url" => @url})
    else
      Output.info("Opening #{@url}")

      cmd =
        case :os.type() do
          {:unix, :darwin} -> "open"
          {:win32, _} -> "start \"\""
          _ -> "xdg-open"
        end

      case System.cmd(cmd, [@url], stderr_to_stdout: true) do
        {_, 0} -> :ok
        _ -> Output.info("Could not open browser. Visit: #{@url}")
      end
    end
  end
end

File: python/src/formhandle/commands/open.py
"""Open API docs in browser."""

import subprocess
import sys
import webbrowser

from formhandle.output import info, json_out

URL = "https://formhandle.dev/swagger/"


def run(args) -> None:
    if args.json:
        json_out({"url": URL})
        return

    info(f"Opening {URL}")
    try:
        webbrowser.open(URL)
    except Exception:
        if sys.platform == "darwin":
            cmd = ["open", URL]
        elif sys.platform == "win32":
            cmd = ["start", "", URL]
        else:
            cmd = ["xdg-open", URL]
        try:
            subprocess.Popen(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
        except Exception:
            info(f"Could not open browser. Visit: {URL}")
